meta-error-handling/static_assert.md

```diff
@@ -0,0 +1,113 @@
+# Module name: `static_assert`
+
+## Overview
+
+`static_assert` is a compile-time evaluated function that asserts the 
+truth of a supplied predicate, issuing an optional user-supplied error 
+message if the predicate is `false`.
+
+<table>
+  <thead>
+    <th>Level</th>
+    <th>Objectives</th>
+  </thead>
+  <tr>
+    <td>Foundational</td>
+    <td>Calling <code>static_assert</code> with a constant expression</td>
+  </tr>
+  <tr>
+    <td>Main</td>
+    <td>Using <code>static_assert</code> to detect contract violations and improve error messages</td>
+  </tr>
+  <tr>
+    <td>Advanced</td>
+    <td></td>
+  </tr>
+</table>
+
+## Motivation
+
+`static_assert` allows the developer to enforce that conditions which can 
+be checked during compilation will force build errors when violated. 
+Additionally, they are the best mechanism by which a developer can pass 
+useful information to other developers regarding what violation occured or 
+what must be done, instead.
+
+## Topic introduction
+
+## Foundational: Calling `static_assert` with a constant expression
+
+### Background/Required Knowledge
+
+A student:
+
+* Should be able to explain the difference between code evaluated at compile-time and run-time
+* Should be able to cite some examples of compile-time known information, such as `sizeof(T)`
+
+### Student outcomes
+
+A student should be able to:
+
+1. Assert the expected size of a structure using `static_assert`
+
+### Caveats
+
+### Points to cover
+
+* In addition to what is wrong, a good error message will inform the user of how to correct it
```

Comment:
```suggestion
* X
* In addition to what is wrong, a good error message will inform the user of how to correct it
```
Maybe, we should add a point here that is related to the `A student should be able to` `1.`